backtrace-library/src/main/java/backtraceio/library/models/json/BacktraceAttributes.java

```diff
@@ -211,41 +203,12 @@ private void convertReportAttributes(BacktraceReport report) {
         }
     }
 
-    /**
-     * Divide custom user attributes into primitive and complex attributes and add to this object
-     *
-     * @param attributes client's attributes
-     */
-    private void convertAttributes(Map<String, Object> attributes) {
-        for (Map.Entry<String, Object> entry : attributes.entrySet()) {
-            Object value = entry.getValue();
-            if (value == null) {
-                continue;
-            }
-            Class type = value.getClass();
-            if (TypeHelper.isPrimitiveOrPrimitiveWrapperOrString(type)) {
-                this.attributes.put(entry.getKey(), value.toString());
-            } else {
-                this.complexAttributes.put(entry.getKey(), value);
-            }
-        }
-    }
-
-    public String getApplicationName() {
-        return this.context.getApplicationInfo().loadLabel(this.context
-                .getPackageManager()).toString();
+    private void convertAttributes(Map<String, Object> clientAttributes) {
+        Tuple<Map<String, String>, Map<String, Object>> reportData = ReportDataBuilder.getReportAttribues(clientAttributes);
+        this.attributes.putAll(reportData.first);
```

Comment:
a null check here would be great

Reply:
null check to verify if the first value is defined? It's always defined I think?

Reply:
Added a null checks to be sure we're not adding nullable/empty client attributes.